origins_subtheme/src/Command/OriginsSubthemeCommand.php

```diff
@@ -0,0 +1,279 @@
+        'Please provide a name for the subtheme',
+        null
+      );
+    }
+
+    // Saves Friendly Input name.
+    $this->friendly_name = $name;
+
+    // Stores argument as machine friendly name.
+    $input->setArgument('name', $this->machine_name_validation($name)); // Validation function to sanitize input.
+
+    // Parent Theme.
+    while(is_null($parent_theme)) {
+      $parent_theme = $this->Io->choice(
+        'Select which Theme to Subtheme off.',
+        array_keys($this->parent_themes)
+      );
+    }
+    $input->setOption('parent_theme', $parent_theme); // Machine name set as value.
+
+    // Set Description (OPTIONAL)
+    $description = $this->Io->askEmpty(
+      'Please provide a description for the subtheme',
+      'Subtheme of ' . $parent_theme
+    );
+    $input->setOption('description', $description);
+    $this->description = $description;
+  }
+
+  /**
+   * Configure functionality for command.
+   */
+  protected function configure() {
+    $this->setName('origins:subtheme')
+      ->setDescription('Automated Subtheme command')
+      ->addArgument(
+      'name',
+      InputArgument::REQUIRED,
+      'Name for subtheme',
+      null)
+      ->addOption('parent_theme', '', InputOption::VALUE_REQUIRED,
+        'Parent Theme to subtheme off.')
+      ->addOption('description', '', InputOption::VALUE_OPTIONAL,
+        'Description for new theme.');
+  }
+
+  /**
+   * Main body of code to execute for command.
+   *
+   * @param InputInterface $input
+   * @param OutputInterface $output
+   * @return int|void|null
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    //copy directory contents.
+    try {
+      $this->filesystem->mirror(
+        self::BASE_THEME_DIRECTORY . $input->getOption('parent_theme').'/' . self::PLACEHOLDER,
+        self::BASE_THEME_DIRECTORY . $input->getArgument('name')
+      );
+
+      $this->iterate_directory($input->getArgument('name'));
+
+    } catch (IOExceptionInterface $exception) {
+      $this->Io->info('Error encountered when making directory at: ' . $exception->getPath());
+    }
+  }
+
+  /**
+   * Cleans up name to be machine friendly.
+   *
+   * @param $name
+   * @return mixed
+   */
+  public function machine_name_validation($name)
+  {
+    // Clean up the machine name.
+    $machine_name = str_replace(' ', '_', strtolower($name));
+
+    // Blacklisted characters.
+    $search = array(
+      '/[^a-z0-9_]/',
+      '/^[^a-z]+/',
+    );
+
+    // Replaces restricted characters.
+    $machine_name = preg_replace($search, '', $machine_name);
+
+    // Returns validated machine name for subtheme.
+    return $machine_name;
+  }
+
+  /**
+   * Returns all Themes with Startkit Directory placeholder
+   *
+   * @return array
+   */
+  public function get_startkit_themes()
+  {
+    // Init Var.
+    $parentThemeOptions = [];
+
+    // Loads all Installed Themes.
+    $themes = \Drupal::service('theme_handler')->listInfo();
```

Comment:
You should use dependency injection here, as a general rule Drupal services shouldn't be called directly from a class method